cnd/src/swap_protocols/rfc003/bob/mod.rs

```diff
@@ -21,9 +24,9 @@ pub struct State<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset> {
     pub swap_communication: SwapCommunication<AL, BL, AA, BA>,
     pub alpha_ledger_state: LedgerState<AL>,
     pub beta_ledger_state: LedgerState<BL>,
-    #[derivative(Debug = "ignore")]
-    pub secret_source: Arc<dyn SecretSource>,
-    pub secret: Option<Secret>,
+    #[derivative(Debug = "ignore", PartialEq = "ignore")]
+    pub secret_source: Seed, // Used to derive identities for refund/redeem.
```

Comment:
:mega: Bob can still derive a secret here. The main idea of the ticket is to make this impossible. To do that, this would need to be something like `dyn DeriveIdentities`.

Reply:
I realise that this PR does in no way inhibit Bob from deriving the secret.  I played with using dyn and could not get it to work because there are times when we must pass a parameter that is for use by Alice _and_ Bob, hence needs `DeriveSecret`.  I could come up with no way in code to separate Bob and Alice call chains since in many other places in the code base there is no explicit separation.  Happy to be proved wrong.

Reply:
It is unfortunate if it doesn't work (yet).

I think, making it work would probably result in a better model because it resembles the real world better (Bob never derives a secret). If you are keen, we can pair-program on this next week and see if we can make progress!